src/connector.ts

```diff
@@ -668,27 +659,29 @@ async function standaloneConnect(
 export default {
   connectNonProjectREPLCommand: async (context: vscode.ExtensionContext) => {
     status.updateNeedReplUi(true);
+    let projectDirUri = state.getProjectRootUri();
+    if (!projectDirUri) {
+      projectDirUri = await state.getOrCreateNonProjectRoot(context, true);
+    }
```

Comment:
it looks like `getOrCreateNonProjectRoot` calls `getProjectRootUri` internally, first thing, and returns it later. Could we ditch the `getProjectRootUri` here and just call `getOrCreateNonProjectRoot` on its own here? Or should we remove the `getProjectRootUri` from `getOrCreateNonProjectRoot`? Or just leave everything as-is?

Reply:
I'll have to check and think. You are probably right. This has been somewhat cleaned up, but I might have missed this opportunity. Tomorrow, though, because right now too 💤 !